py/tests/release.py

```diff
@@ -0,0 +1,192 @@
+    parser.add_argument('--maxheight', type=int, default=10,
+                        help='Number of blocks to mine')
+    parser.add_argument('--tarball', required=True, 
+                        help='Release package tarball')
+
+    parser.add_argument('--version', required=True, 
+                        help='Release package version')
+
+    args = parser.parse_args()
+    tar_file_name = args.tarball
+    max_height = args.maxheight
+    return (tar_file_name, max_height, args.version)
+
+def tail_logs(temp_dir, log_name):
+    n = 200 # last 200 lines
+    f = os.path.join(temp_dir, "log", log_name)
+    stdin, stdout = os.popen2("tail -n "+ str(n) + " " + f)
+    stdin.close()
+    lines = "\n".join(stdout.readlines())
+    stdout.close()
+    return lines
+
+
+def setup_node(node, path, version):
+    os.chdir(path)
+    for command in NODE_SETUP_COMMANDS: 
+        os.system(pystache.render(command, {"version": version, \
+                                            "name": SETUP[node]["name"]}))
+    file_obj = open(os.path.join(path, "p.config"), "w")
+    file_obj.write(SETUP[node]["config"])
+    file_obj.close()
+
+def main(argv):
+    tar_file_name, max_height, version = read_argv(argv)
+    root_dir = tempfile.mkdtemp()
+    temp_dir_dev1 = os.path.join(root_dir, "node1") 
+    os.makedirs(temp_dir_dev1)
+
+    temp_dir_dev2 = os.path.join(root_dir, "node2") 
+    temp_dir_dev3 = os.path.join(root_dir, "node3") 
+
+
+    print("Tar name: " + tar_file_name)
+    extract_tarball(tar_file_name, temp_dir_dev1)
+    shutil.copytree(temp_dir_dev1, temp_dir_dev2)
+    shutil.copytree(temp_dir_dev1, temp_dir_dev3)
+
+    node_names = ["node1", "node2", "node3"]
+    node_dirs = [temp_dir_dev1, temp_dir_dev2, temp_dir_dev3]
+    [setup_node(n, d, version) for n, d in zip(node_names, node_dirs)]
+    [start_node(d) for d in node_dirs]
+
+    time.sleep(30)
+    node_objs = [ExternalApi(ApiClient(host=SETUP[n]["host"])) for n in node_names]
+    height = 0
+    test_failed = False
+    try:
+        print("Will mine till block " +  str(max_height))
+        while height < max_height:
+            time.sleep(1) # check every second
+            for name, node in zip(node_names, node_objs):
+                top = node.get_top() # node is alive and mining
+                print("[" + name + "] height=" + str(top.height))
+                height = max(height, top.height)
+            print("")
+    except ApiException as e:
+        test_failed = True
+        print("node died")
+    except urllib3.exceptions.MaxRetryError as e:
+        test_failed = True
+        print("node died")
+    [stop_node(d) for d in node_dirs]
+    if test_failed:
+        for name, node_dir in zip(node_names, node_dirs):
+            print(name + " logs:")
+            print(tail_logs(node_dir, "epoch.log"))
+            print("\n")
+    shutil.rmtree(root_dir)
+    if test_failed:
+        sys.exit("FAILED")	
```

Comment:
I see [this results in exit code 1](https://docs.python.org/3/library/sys.html#sys.exit). Ok.